src/redux/actions/passwordResetActions.js

```diff
@@ -0,0 +1,59 @@
+import axios from 'axios';
+import {
+  PASSWORD_REQUEST_LOADING, PASSWORD_REQUEST_SUCCESS, PASSWORD_REQUEST_FAILED, INITIAL_STATE,
+} from './types/resetPasswordType';
+
+export const requestLoading = () => ({
+  type: PASSWORD_REQUEST_LOADING,
+  payload: {
+    loading: true,
+  },
+});
+
+export const success = message => ({
+  type: PASSWORD_REQUEST_SUCCESS,
+  payload: {
+    loading: false,
+    success: true,
+    message,
+  },
+});
+
+export const failed = message => ({
+  type: PASSWORD_REQUEST_FAILED,
+  payload: {
+    loading: false,
+    message,
+  },
+});
+
+export const initialState = () => ({
+  type: INITIAL_STATE,
+  payload: {
+    loading: false,
+    success: false,
+    message: '',
+  },
+});
+
+export const requestPasswordResetLink = email => async (dispatch) => {
+  try {
+    dispatch(requestLoading());
+    const result = await axios.post('https://mazus-ah-staging.herokuapp.com/api/v1/auth/forgotpassword', { email });
+    dispatch(success(result.data.auth.message));
+  } catch (err) {
+    dispatch(failed(err.response.data.errors.message));
+    setTimeout(() => { dispatch(initialState()); }, 2000);
```

Comment:
This is an example of where I'm using the initialState action to restore the page 2 seconds after an error is shown.